util/kvdb-rocksdb/src/lib.rs

```diff
@@ -491,6 +491,9 @@ impl Database {
 				let batch = WriteBatch::new();
 				let ops = tr.ops;
 				for op in ops {
+					// remove any buffered operation for this key
+					self.overlay.write()[Self::to_overlay_column(op.col())].remove(op.key());
```

Comment:
Not sure if I should take a read lock first to check if there's any entry, instead of always locking for write.

Reply:
No sure either; is there any reliable way to just benchmark this?

Reply:
I believe, we always lock for write